### PR TITLE
AArch64 Add Fiber support and fix GC scanning

### PR DESCRIPTION
**Pull request by smolt**
Add AArch64 Fiber support.  Fix register save code for GC scanning.

@redstar - are you able to test these with your setup?  My testing is limited to iPhone 6.

### CODE REVIEW

File: src/core/thread.d [hunk 1]
```diff
@@ -38,12 +38,6 @@ version( Solaris )
     import core.sys.solaris.sys.types;
 }
 
-version( Solaris )
-{
-    import core.sys.solaris.sys.priocntl;
-    import core.sys.solaris.sys.types;
-}
-
```
Comment (smolt) [inline src/core/thread.d:46]: cleanup duplicate block while here.

File: src/core/thread.d [hunk 2]
```diff
@@ -2463,23 +2457,17 @@ else
...
-            __asm("str x28, $0", "=*m", regs.ptr + 9);
+            __asm("stp x19, x20, $0", "=*m", regs.ptr + 0);
+            __asm("stp x21, x22, $0", "=*m", regs.ptr + 2);
+            __asm("stp x23, x24, $0", "=*m", regs.ptr + 4);
+            __asm("stp x25, x26, $0", "=*m", regs.ptr + 6);
+            __asm("stp x27, x28, $0", "=*m", regs.ptr + 8);
             __asm("str x29, $0", "=*m", regs.ptr + 10);
-
-            __asm("str x31, $0", "=*m", &sp);
+            sp = __asm!(void*)("mov $0, sp", "=r");
```
Comment (smolt) [inline src/core/thread.d]: reg x31 in context of store is zero reg instead of stack pointer.

**Comment by redstar** [comment]
I try to check. Currently my Gentoo VM is broken (crash of Windows host :-( ).

**Comment by joakim-noah** [comment]
Shouldn't this go upstream?  There's nothing specific to ldc here.

**Comment by smolt** [comment]
How about the ldc `__asm()`?  Those don't go upstream to druntime, right?  Those are in LDC blocks - or should be.  I never noticed until now, but I think `callWithStackShell()` should have all the LDC inline asm within an LDC version block.

Would happily make an upstream PR for the rest though.

**Comment by joakim-noah** [comment]
Ah, missed that.  Yeah, should be within an LDC version block and the rest should go upstream.

**Pull request merged**